share/revng/test/tests/scripting/daemon.py

```diff
@@ -0,0 +1,84 @@
+
+from revng.scripting import DaemonProject
+from revng import ptml
+import subprocess
+from time import sleep
+import psutil
+import atexit
+import sys
+import re
+
+
+def exit_handler(pid):
+    process = psutil.Process(pid)
+    for proc in process.children(recursive=True):
+        proc.kill()
+    process.kill()
+
+
+daemon_process = subprocess.Popen(f"REVNG_DATA_DIR='{sys.argv[1]}' orc shell revng daemon", shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+atexit.register(exit_handler, daemon_process.pid)
+sleep(5)
+
+url = "http://127.0.0.1:8000"
+daemon_client = DaemonProject(url, sys.argv[2])
+#daemon_client = DaemonProject(url)
+
+# daemon_client.upload_binary(sys.argv[2])
+# daemon_client.analyses_list("revng-initial-auto-analysis")
+# daemon_client.analyses_list("revng-c-initial-auto-analysis")
+
+daemon_client.model.TypeDefinitions[1].get_artifact("emit-type-definitions")
+
+# Get the first function with `CustomName` and the functions index
+function_original_name = ""
+function_idx = 0
+for idx, function in enumerate(daemon_client.model.Functions):
+    if function.CustomName:
+        function_original_name = function.CustomName
+        function_idx = idx
+        break
+assert function_original_name != ""
+
+# Get artifact by using the project and function, the result should be the same
+result1 = daemon_client.model.Functions[function_idx].get_artifact("disassemble")
+result2 = daemon_client.get_artifact("disassemble", [daemon_client.model.Functions[function_idx]])
+assert ptml.parse(result1) == ptml.parse(result2)
+
+# Make sure that we get the correct function name in the parse result
+re_match = re.search(function_original_name, ptml.parse(result1))
+assert re_match is not None
+
+# Change the function name, get an artifact and make sure that the
+# parsed result contains the new name
+function_new_name1 = "lol"
+daemon_client.model.Functions[function_idx].CustomName = function_new_name1
+result1 = daemon_client.model.Functions[function_idx].get_artifact("disassemble")
+re_match = re.search(function_new_name1, ptml.parse(result1))
+assert re_match is not None
+
+# Change function name again and commit
+function_new_name2 = "lolz"
+daemon_client.model.Functions[function_idx].CustomName = function_new_name2
+daemon_client.commit()
+
+daemon_client = DaemonProject(url, sys.argv[2])
+# Load the project again and run the analysis again
+# daemon_client.upload_binary(sys.argv[2])
+# daemon_client.analyses_list("revng-initial-auto-analysis")
+# daemon_client.analyses_list("revng-c-initial-auto-analysis")
+
+# make sure that the second name changes was commitd
+assert daemon_client.model.Functions[function_idx].CustomName == function_new_name2
+
+# Get multiple artifacts
+result = daemon_client.get_artifacts({
+    "disassemble": [
+        daemon_client.model.Functions[0],
+        daemon_client.model.Functions[1]
+    ],
+    "decompile": []
+})
+ptml.parse(result["disassemble"])
+ptml.parse(result["decompile"])
```

Comment:
The functionality here seems to be greatly duplicated with `cli.py`. I'd do the following:
* Merge the two together in `share/revng/test/tests/scripting.py`
* Handle arguments via `argparse`
* Add the options `--cli` and `--daemon`, e.g.
   ```
   ./scripting.py --cli $DATA_DIR $INPUT
   ./scripting.py --daemon $DATA_DIR $INPUT
   ```